### PR TITLE
add python support for sets & maps

### PR DESCRIPTION
**Pull request by ryanchou-dev**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: content/2_Bronze/Intro_Sets.mdx [hunk 1]
```diff
@@ -1226,4 +1226,222 @@ make their implementation easier.
...
+		<Quiz.Answer correct>
+			$\mathcal{O}(\log N)$
+			<Quiz.Explanation>
+				Correct.
+			</Quiz.Explanation>
+		</Quiz.Answer>
+	</Quiz.Question>
+	<Quiz.Question>
+		Which of the following is the correct output of this code?
+		```py
```
Comment (SansPapyrus683) [inline content/2_Bronze/Intro_Sets.mdx:1254]: are languagesections not supported in quiz questions?

Comment (ryanchou-dev) [reply content/2_Bronze/Intro_Sets.mdx:1254]: image

nope, messes up answer selection cuz they all have diff questions and they must be directly under the question comp

+ it's prolly easier to edit if the quizzes are separated by language 

—--

you also can't have language sections underneath a <quiz> tag

Comment (SansPapyrus683) [reply content/2_Bronze/Intro_Sets.mdx:1254]: fff alright then

File: content/2_Bronze/Intro_Sets.mdx [hunk 2]
```diff
@@ -1226,4 +1226,222 @@ make their implementation easier.
...
+		s = set()
+		s.add(1)
+		s.add(4)
+		s.add(5)
+		s.add(2)
```
Comment (SansPapyrus683) [inline content/2_Bronze/Intro_Sets.mdx:1259]: why not just `s = {1, 4, 5, 2}`

File: content/2_Bronze/Intro_Sets.mdx [hunk 3]
```diff
@@ -1226,4 +1226,222 @@ make their implementation easier.
...
+			<Quiz.Explanation>
+Incorrect. The output is a `bool`.
+			</Quiz.Explanation>
+		</Quiz.Answer>
+
+    </Quiz.Question>
+    <Quiz.Question>
+    	Which of the following is the correct output of this code?
+    	```py
+    	s = set([4, 1, 2, 1, 3, 2, 3, 1, 3, 1])
```
Comment (SansPapyrus683) [inline content/2_Bronze/Intro_Sets.mdx:1291]:
```suggestion
    	s = {4, 1, 2, 1, 3, 2, 3, 1, 3, 1}

File: content/2_Bronze/Intro_Sets.mdx [hunk 4]
```diff
@@ -1226,4 +1226,222 @@ make their implementation easier.
...
+
+    </Quiz.Question>
+    <Quiz.Question>
+    	Which of the following is the correct output of this code?
+    	```py
+    	m = dict()
+    	m[1] = 5
+    	m[2] = 9
+    	m[3] = 7
+    	m[4] = 3
```
Comment (SansPapyrus683) [inline content/2_Bronze/Intro_Sets.mdx:1377]: why not direct initialization

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/2_Bronze/Intro_Sets.mdx (hunk 4 above)
Comment (ryanchou-dev) [reply content/2_Bronze/Intro_Sets.mdx:1377]: keeping consistent between langs + familiarizing with mutating dictionaries

File: content/2_Bronze/Intro_Sets.mdx (hunk 3 above)
Comment (ryanchou-dev) [reply content/2_Bronze/Intro_Sets.mdx]: NOO IM GETTING RUSTY IN PYTHON 😭

File: content/2_Bronze/Intro_Sets.mdx (hunk 2 above)
Comment (ryanchou-dev) [reply content/2_Bronze/Intro_Sets.mdx:1259]: similar reason to: https://github.com/cpinitiative/usaco-guide/pull/3677#discussion_r1199823416

File: content/2_Bronze/Intro_Sets.mdx [hunk 5]
```diff
@@ -1226,4 +1226,222 @@ make their implementation easier.
...
+			True
+			<Quiz.Explanation>
+				Incorrect. $3$ isn't present in `s`.
+			</Quiz.Explanation>
+		</Quiz.Answer>
+		<Quiz.Answer>
+			$0$
+			<Quiz.Explanation>
+Incorrect. The output is a `bool`.
+			</Quiz.Explanation>
```
Comment (SansPapyrus683) [inline content/2_Bronze/Intro_Sets.mdx:1284]: sussy indentation

File: content/2_Bronze/Intro_Sets.mdx [hunk 6]
```diff
@@ -1226,4 +1226,222 @@ make their implementation easier.
...
+    	<Quiz.Answer correct>
+    		$3$
+    		<Quiz.Explanation>
+    			Correct. `m[2]` is changed from $5$ to $3$.
+    		</Quiz.Explanation>
+    	</Quiz.Answer>
+    	<Quiz.Answer>
+    		$5$
+    		<Quiz.Explanation>
+    			Note that while `m[2]` is set to $5$ at first, it is changed later on.
```
Comment (SansPapyrus683) [inline content/2_Bronze/Intro_Sets.mdx:1433]:
```suggestion
    			While `m[2]` *is* set to $5$ at first, it's changed later on.
```

**Review by SansPapyrus683 — APPROVED** [review]

**Comment by SansPapyrus683** [comment]
closes #3675

File: content/2_Bronze/Intro_Sets.mdx [hunk 7]
```diff
@@ -1226,4 +1226,222 @@ make their implementation easier.
...
+    		$7$
+    		<Quiz.Explanation>
+
+    		</Quiz.Explanation>
+    	</Quiz.Answer>
+    	<Quiz.Answer>
+    		$10$
+    		<Quiz.Explanation>
+    	Sets only store unique elements.
+    		</Quiz.Explanation>
```
Comment (SansPapyrus683) [inline content/2_Bronze/Intro_Sets.mdx:1316]: bad indentation here

File: content/2_Bronze/Intro_Sets.mdx [hunk 8]
```diff
@@ -1226,4 +1226,222 @@ make their implementation easier.
...
+    		`KeyError: 2`
+    		<Quiz.Explanation>
+    			Correct. $2$ is no longer associated with a value.
+    		</Quiz.Explanation>
+    	</Quiz.Answer>
+    	<Quiz.Answer>
+    		$9$
+    		<Quiz.Explanation>
+    $m[2]$ is erased, so $m[2]$ is no longer $9$.
+    		</Quiz.Explanation>
```
Comment (SansPapyrus683) [inline content/2_Bronze/Intro_Sets.mdx:1391]: here

Comment (ryanchou-dev) [reply content/2_Bronze/Intro_Sets.mdx:1391]: ah so sorry

**Review by envyaims — APPROVED** [review]
e

**Pull request merged**